src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1660,7 +1660,15 @@ tx_gateway_frontend::do_end_txn(
 
     checked<cluster::tm_transaction, tx_errc> r(tx_errc::unknown_server_error);
     if (request.committed) {
-        if (tx.status == tm_transaction::tx_status::ongoing) {
+        bool is_status_ok = false;
+        if (is_fetch_tx_supported()) {
```

Comment:
looks like the new RPC was added for this purpose, this needs some documentation.